lib/java/com/google/android/material/internal/CollapsingTextHelper.java

```diff
@@ -447,6 +449,11 @@ public void apply(Typeface font) {
     recalculate();
   }
 
+  public void setTitleTextEllipsize(TruncateAt ellipsize) {
+    titleTextEllipsize = ellipsize;
+    recalculate();
```

Comment:
Did you test if this will work if you display the collapsing toolbar first and then change the ellipsize? I think we need some logic in calculateUsingTextSize() to set updateDrawText when ellipsize changes.

Reply:
Yes, I have tested the below use cases.

1. set Ellipsize and title in xml layout -- Passed
2. set Ellipsize as end in xml layout and then change the ellipsize to middle -- Passed
3. set Ellipsize as end and title text in xml layout and then change title, ellipsize to middle -- Passed 

//Code snipped I used in TopAppBarCollapsingLargeMiddleEllipsizeDemoFragment to verify below

  @Override
  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    CollapsingToolbarLayout layout = view.findViewById(R.id.collapsingtoolbarlayout);
    layout.setTitle("Large Collapsing Ellipsize Toolbar Title Ellipsize test test");
    layout.setTitleEllipsize(TextUtils.TruncateAt.MIDDLE);
  }